tasks/configure-files.yml

```diff
@@ -1,12 +1,28 @@
 ---
-- name: Configure JCasC configuration file
+# We need to make a subdirectory for JCasC configuration files, since the JCasC plugin
+# examines all YAML files in this directory and we'll also put the plugins.yaml file here,
+# which isn't meant for JCasC.
+- name: Ensure JCasC configuration subdirectory exists
+  become: true
+  ansible.builtin.file:
+    path: "{{ jenkins_config_dir }}/jcasc/"
+    owner: "{{ jenkins_user }}"
+    group: "{{ jenkins_group }}"
+    mode: "0755"
+    state: directory
+
+- name: Configure JCasC configuration files
   become: true
   ansible.builtin.template:
-    src: "{{ jenkins_jcasc_config_file }}"
-    dest: "{{ jenkins_config_dir }}/jenkins.yaml"
+    src: "{{ config_file }}"
+    dest: >-
+      {{ jenkins_config_dir }}/jcasc/{{ config_file | basename | replace('.j2', '') }}
```

Comment:
Can we make the `replace` call a bit more specific?

Reply:
Sure! Fixed in 5f1193d.